Autopilot/Src/ultrasonic.cpp

```diff
@@ -0,0 +1,63 @@
+#include "ultrasonic.hpp"
+#include "stm32f765xx.h"
+#include "stm32f7xx_hal_tim.h"
+#include "stm32f7xx_hal_gpio.h"
+#include "tim.h"
+#include <cstdint>
+
+#define SPEED_OF_SOUND 0.034f  
+#define TRIG_PIN   5 // temporary value, to be changed when pin configs completed
+
+uint32_t IC_Val1 = 0;
+uint32_t IC_Val2 = 0;
+uint32_t difference = 0;
+uint8_t isFirstCaptured = 0; // Is the first value captured
+uint8_t distance = 0; // computed distance based on the length of the ECHO 
+
+HCSR04 :: HCSR04() {
+    HAL_TIM_IC_Start_IT(&htim11, TIM_CHANNEL_1); // Starts the timer (to be changed after pin configuration)
+    data.htim = htim11;
+}
+
+void HCSR04::trigger(GPIO_TypeDef * triggerPin, uint32_t pinName) {
+    triggerPin -> BSRR |= (1 << TRIG_PIN); // set GPIO pin high
+    this->delayMicroseconds(10); // Delay/wait for 10 us as defined in datasheet
+    triggerPin -> BSRR |= (1 << (16 + TRIG_PIN)); // reset the pin, hence completing the sending of the pulse
+}
+
+void HCSR04::delayMicroseconds(uint32_t us) {
+    int16_t currentTim = htim11.Instance -> CNT;
+    while ((htim11.Instance -> CNT - currentTim) < us); // wait the required microseconds before ending
+}
+
+void HCSR04 :: getDistance(ultrasonicData_t * data) {
+    trigger(data -> triggerPin, TRIG_PIN); // Sends trigger pulse 
+    __HAL_TIM_ENABLE_IT(data -> htim, TIM_IT_CC1);
+}
```

Comment:
This function feels misleading, it doesn't actually update the data, just triggers the measurement. I'd expect getDistance to update the data, please clarify in comment or rename function